prow/cmd/deck/tide.go

```diff
@@ -147,68 +151,118 @@ func (ta *tideAgent) updateHistory() error {
 	if err := fetchTideData(ta.log, path, &history); err != nil {
 		return err
 	}
-	history = ta.filterHiddenHistory(history)
+	history = ta.filterHistory(history)
 
 	ta.Lock()
 	defer ta.Unlock()
 	ta.history = history
 	return nil
 }
 
-func (ta *tideAgent) filterHiddenPools(pools []tide.Pool) []tide.Pool {
-	if len(ta.hiddenRepos()) == 0 {
-		return pools
-	}
+func (ta *tideAgent) matchingIDs(ids []string) bool {
+	return len(ids) > 0 && sets.String{}.Insert(ta.tenantIDs...).HasAll(ids...)
+}
 
+func (ta *tideAgent) filterPools(pools []tide.Pool) []tide.Pool {
 	filtered := make([]tide.Pool, 0, len(pools))
 	for _, pool := range pools {
-		needsHide := matches(pool.Org+"/"+pool.Repo, ta.hiddenRepos())
-		if needsHide && ta.showHidden {
-			filtered = append(filtered, pool)
-		} else if needsHide == ta.hiddenOnly {
+		// curIDs are the IDs associated with all PJs in the Pool
+		// We want to add the ID associated with the OrgRepo for extra protection
+		curIDs := sets.NewString(pool.TenantIDs...)
+		orgRepoID := ta.cfg.GetProwJobDefault(pool.Org+"/"+pool.Repo, "*").TenantID
+		// If the orgrepo is associated with no tenantID OR the default tenantID we ignore it here.
+		// This prevents already IDd pools from getting the default ID assigned to them when their orgrepo is not associated with an OrgRepo.
+		// Pools with no tenantID and with default tenantID behave the same, so adding the default ID just causes issues
```

Comment:
> Pools with no tenantID and with default tenantID behave the same

I think this should be true, but it isn't currently. Currently the `matchingIDs` function always returns false if the list of ids is empty. If the list were to contain the default ID it could match and return true.

Also what issues are you referring to?